.circleci/config.yml

```diff
@@ -51,6 +51,57 @@ jobs:
       - run: wget -O ./bin/manifest-tool https://github.com/estesp/manifest-tool/releases/download/v1.0.0/manifest-tool-linux-amd64 && chmod +x ./bin/manifest-tool
       - run: echo "export GO111MODULE=on" >>$BASH_ENV
       - run: PATH=./bin:$PATH inv push-multiarch --binaries=speaker --tag=${CIRCLE_BRANCH:-${CIRCLE_TAG}} --docker-user=metallb
+  deploy-controller-pr:
+    working_directory: /go/src/go.universe.tf/metallb
+    docker:
+      - image: circleci/golang:1.13
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run: sudo apt-get install python-pip
+      - run: sudo pip install invoke semver pyyaml
+      - run: docker login -u $DOCKER_USER -p $DOCKER_PASSWORD
+      - run: mkdir -p ./bin
+      - run: wget -O ./bin/manifest-tool https://github.com/estesp/manifest-tool/releases/download/v1.0.0/manifest-tool-linux-amd64 && chmod +x ./bin/manifest-tool
+      - run: echo "export GO111MODULE=on" >>$BASH_ENV
+      - run: echo ${CIRCLE_PULL_REQUEST}
+      - run: echo ${CIRCLE_BRANCH}
+      - run: echo ${CIRCLE_TAG}
+      # CIRCLE_PULL_REQUEST is a URL in the form: https://github.com/metallb/metallb/pull/1
+      # We use sed to get the pull request ID out of the URL.  CIRCLE_PR_NUMBER would be
+      # more convenient, but unfortunately it is not always available.
+      # https://circleci.com/docs/2.0/env-vars/
+      #
+      # If the job is being run against a PR, use "pr-<ID>" as the image tag and push the
```

Comment:
I guess CircleCI doesn't support triggering only on PRs (which would make this check unnecessary). I see we use branch filters but we still need this check in case someone (e.g. me :grimacing:) pushes to a branch in this repo (right?).

Reply:
Right ... it may be possible, but I haven't figured out how.  I had to change a previous attempt to filter out pull requests here: https://github.com/metallb/metallb/commit/8b5c60f55a7799cec01e15b94cf6b7bc3dc8ad34